fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintenanceTest.java

```diff
@@ -698,7 +712,12 @@ public void close() throws IOException {
 
                 @Override
                 public void ensureValid() throws IOException {
-                    throw new IOException("invalid lock");
+                    // 0 <= 0-99 < 100
+                    if (ThreadLocalRandom.current().nextInt(100) < percentFailure) {
```

Comment:
We could make the test more deterministic if the test took a seed as a parameter, and created a `Random` for this.